### ISSUE TITLE
Update validation function names (proposal)

### ISSUE THREAD

**Issue by rmarren1** [plotly/dash#413]
I think we should update a few function names in `dash.py` that might cause some confusion when the component validation PR is released.

`dash._validate_callback` runs a bunch of tests _at the time a user defines a callback_, such as:
* checking that the layout has already been defined
* checking that the output ID and input IDs exist in the layout
* checking that properties actually exist in the components referenced by ID in the callback

I think this should be called `dash._validate_callback_definition` to be more specific that this function is validating the definition of a callback rather than the output of the callback itself.

`dash._validate_callback_output` is run after an attempt to serialize the output of a callback fails as a debug step. I named this before working on the component validation PR, and now I think it would cause confusion since validation of callback outputs is done in another function entirely. I would like to move the function that _actually_ validates callback output to `dash._validate_callback_output`, and change the previous function to `dash._debug_callback_serialization_error` since this function is only run to debug a callback serialization error.

In summary, the following changes:
* `dash._validate_callback` --> `dash._validate_callback_definition`
* `dash._validate_callback_output` --> `dash._debug_callback_serialization_error`
* _logic to validate callback output against schema_ --> `dash._validate_callback_output`

@plotly/dash Does this nomenclature make sense?

**Comment by chriddyp** [comment]
👍 makes sense to me

**Comment by T4rk1n** [comment]
These methods are "private", feel free to rename them if it makes more sense 👍
